compiler/resolution/functionResolution.cpp

```diff
@@ -5506,19 +5564,21 @@ static void resolveMoveForRhsSymExpr(CallExpr* call) {
     Symbol* lhsSym  = toSymExpr(call->get(1))->symbol();
     Type*   rhsType = rhs->typeInfo();
 
-    INT_ASSERT(lhsSym->hasFlag(FLAG_INDEX_VAR) ||
-               // non-zip forall over a standalone iterator
-               rhs->symbol()->hasFlag(FLAG_INDEX_VAR));
-
-    // ... and not of a reference type
-    // ... and not an array (arrays are always yielded by reference)
-    // todo: differentiate based on ref-ness, not _ref type
-    // todo: not all const if it is zippered and one of iterators is var
-    if (isReferenceType(rhsType)                == false &&
-        isTupleContainingAnyReferences(rhsType) == false &&
-        rhsType->symbol->hasFlag(FLAG_ARRAY)    == false) {
-      // ... then mark LHS constant.
-      lhsSym->addFlag(FLAG_CONST);
+    if (lhsSym->hasFlag(FLAG_INDEX_VAR) ||
+        // non-zip forall over a standalone iterator
+        (lhsSym->hasFlag(FLAG_TEMP) &&
+         rhs->symbol()->hasFlag(FLAG_INDEX_VAR))) {
+
+      // ... and not of a reference type
+      // ... and not an array (arrays are always yielded by reference)
+      // todo: differentiate based on ref-ness, not _ref type
+      // todo: not all const if it is zippered and one of iterators is var
+      if (isReferenceType(rhsType)                == false &&
+          isTupleContainingAnyReferences(rhsType) == false &&
+          rhsType->symbol->hasFlag(FLAG_ARRAY)    == false) {
+        // ... then mark LHS constant.
+        lhsSym->addFlag(FLAG_CONST);
```

Comment:
Since the lhs is a temporary in the case where only the rhs is a index variable, does that mean the origin of the lhs temporary is not getting marked const when it should be?